selftests/check.py

```diff
@@ -727,6 +727,24 @@ def main(args):  # pylint: disable=W0621
         "optional-plugins": False,
     }
 
+    test_size = {
+        "static-checks": 7,
+        "job-api-1": 1,
+        "job-api-2": 1,
+        "job-api-3": 2,
+        "job-api-4": 9,
+        "job-api-5": 12,
+        "job-api-6": 4,
+        "job-api-7": 1,
+        "nrunner-interface": 100,
```

Comment:
The size of the suites will vary depending on the availability of Python modules (or may do so because of other aspects).  We have to account for that.  Within the [building of RPMs](https://download.copr.fedorainfracloud.org/results/packit/avocado-framework-avocado-5822/centos-stream-8-x86_64/06724312-python-avocado/builder-live.log.gz) we get:

```
RESULTS    : PASS 1057 | ERROR 0 | FAIL 0 | SKIP 110 | WARN 0 | INTERRUPT 0 | CANCEL 4
JOB HTML   : /builddir/avocado/job-results/job-2023-12-04T16.54-5898053/results.html
JOB TIME   : 285.57 s
suite nrunner-interface doesn't have 100 testsit has 80.
If you made some changes into selftests please update `test_size` variable in `check.py`. If you haven't done any changes to selftests this behavior is an ERROR, and it needs to be fixed.
suite optional-plugins doesn't have 98 testsit has 95.
If you made some changes into selftests please update `test_size` variable in `check.py`. If you haven't done any changes to selftests this behavior is an ERROR, and it needs to be fixed.
```

Also notice the lack of a space between "tests" and "it".

Reply:
Because we have the information that is used to either include tests (or not) in the suite, we can use it to calculate the expected size of the test suite.  It just requires a bit a bit of extra logic at the checks.

Leaving it to CI only would likely mean that contributors will push seltests changes, then watch the CI fail, then address it, then push it again.  It's preferable, when it's cheap, to make the local `make check` experience catch issues too.